a10_octavia/controller/worker/tasks/nat_pool_tasks.py

```diff
@@ -0,0 +1,116 @@
+#    Copyright 2020, A10 Networks
+#
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+
+from oslo_log import log as logging
+from taskflow import task
+
+import acos_client.errors as acos_errors
+
+from a10_octavia.controller.worker.tasks.decorators import axapi_client_decorator
+
+LOG = logging.getLogger(__name__)
+
+
+class NatPoolCreate(task.Task):
+    """Task to create nat pool"""
+
+    @axapi_client_decorator
+    def execute(self, loadbalancer, vthunder, flavor_data=None):
+        device_pool = None
+        if flavor_data:
+            natpool_flavor_list = flavor_data.get('nat_pool_list')
+            natpool_flavor = flavor_data.get('nat_pool')
+            if natpool_flavor_list:
+                for i in range(len(natpool_flavor_list)):
```

Comment:
There isn't a reason to iterate over the indices. 

`for nat_pool in natpool_flavor_list ` results in more readable succinct code

Reply:
Here it is iterated over the indices because natpool_flavor_list is a list of dictionaries in the form shown below:
`"nat-pool-list":[{"pool-name":"pool2", "start-address":"172.16.2.101", "end-address":"172.16.2.102", "netmask":"/24", "gateway":"172.16.2.1"}, {"pool-name":"pool3", "start-address":"172.16.3.101", "end-address":"172.16.3.102", "netmask":"/24", "gateway":"172.16.3.1"}]`

Reply:
Right so if natpool_flavor_list has a list of dictionaries then each "nat_pool" obtained in the loop will hold a dictionary

Reply:
Ok, making this change